python/pyspark/pandas/generic.py

```diff
@@ -1312,11 +1326,20 @@ def sum(psser: "Series") -> Column:
             return F.coalesce(F.sum(spark_column), SF.lit(0))
 
         return self._reduce_for_stat_function(
-            sum, name="sum", axis=axis, numeric_only=numeric_only, min_count=min_count
+            sum,
+            name="sum",
+            axis=axis,
+            numeric_only=numeric_only,
+            min_count=min_count,
+            skipna=skipna,
         )
 
     def product(
-        self, axis: Optional[Axis] = None, numeric_only: bool = None, min_count: int = 0
+        self,
+        axis: Optional[Axis] = None,
+        skipna: bool = True,
```

Comment:
nit: if end user call this api without parameter names like this: `.product(None, True)`

then, the behavior will be changed. should we append new parameters at the end?


otherwise, looks good

Reply:
Good point! 
One concern is maintaining pandas-on-Spark backward compatibility may sacrifice API compatibility(parameter order) with pandas.
`skipna`exists in older pandas like 1.0.

CC @HyukjinKwon @ueshin @itholic

Reply:
+1 to keep pandas behavior, we might want to mention this in migration guide?

Reply:
Agree! I will use `..verionchanged` as a mark to remind us of mentioning it in the migration guide from 3.3 to 3.4.